src/Random.savi

```diff
@@ -166,4 +166,14 @@
       @unbiased_u32_less_than(limit.u32).usize
     |
       @unbiased_u64_less_than(limit.u64).usize
-    )
\ No newline at end of file
+    )
+
+  :: Return a UUID version 4 variant 1. Eg: "2e5fda72-856b-436f-ad83-8d42d096b866"
+  :fun ref uuid String
+    String.join([
+      "\(@u32.format.hex.bare)"
+      "\(@u16.format.hex.bare)"
+      "\(@u16.bit_and(0x0fff).bit_or(0x4000).format.hex.bare)"
+      "\(@u16.bit_and(0x3fff).bit_or(0x8000).format.hex.bare)"
+      "\(@u32.format.hex.bare)\(@u16.format.hex.bare)"
+    ], "-")
```

Comment:
@jemc I was hoping for a more succinct way of doing this, but couldn't quite find my way into it.

Something like generating all the random bytes, apply the masks for version and variant, and then `sprintf` in hex format with the `-` in the right places.

Reply:
this actually points out a limitation of string interpolation

Reply:
I think what we want is a something like this:

```savi
<<-EOS
     \(@u32.format.hex.bare)
    -\(@u16.format.hex.bare)
    -\(@u16.bit_and(0x0fff).bit_or(0x4000).format.hex.bare)
    -\(@u16.bit_and(0x3fff).bit_or(0x8000).format.hex.bare)
    -\(@u32.format.hex.bare)\(@u16.format.hex.bare)
EOS
```

A multi-line interpolated string literal that ignores leading (and trailing) whitespace.

Reply:
I don't think we need any special/new syntax for line breaks in interpolated strings because the line breaks can just be placed inside the parentheses.

For example, this is how I would write the above code, without needing new syntax, and without resorting to `String.join`:

```savi
```suggestion
    ], "-")
```
"\(
  @u32.format.hex.bare
)-\(
  @u16.format.hex.bare
)-\(
  @u16.bit_and(0x0fff).bit_or(0x4000).format.hex.bare
)-\(
  @u16.bit_and(0x3fff).bit_or(0x8000).format.hex.
)-\(
  @u32.format.hex.bare)\(@u16.format.hex.bare
)"
```